front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/buildPacedTrainException.ts

```diff
@@ -173,3 +175,126 @@ export default function updatePacedTrainException(
+    if (
+      exception.constraint_distribution &&
+      isEqual(exception.constraint_distribution.value, updatedPacedTrain.constraint_distribution)
+    ) {
+      delete updatedException.constraint_distribution;
+    }
+
+    if (
+      exception.initial_speed &&
+      isEqual(exception.initial_speed.value, updatedPacedTrain.initial_speed)
+    ) {
+      delete updatedException.initial_speed;
+    }
+
+    if (exception.labels && isEqual(exception.labels.value, updatedPacedTrain.labels)) {
+      delete updatedException.labels;
+    }
+
+    if (exception.options && isEqual(exception.options, updatedPacedTrain.options)) {
+      delete updatedException.options;
+    }
+
+    // Compute first all path steps of the exception and the updated paced train to facilitate the comparison
+    // As the front generates each path step id, between two same pathfinding, ids could be different
+    // so we don't want to compare them.
+    if (updatedException.path_and_schedule) {
+      const originalPacedTrainPathSteps = updatedPacedTrain.path.map((_, i) =>
+        computeBasePathStep(updatedPacedTrain, i)
+      );
+      const exceptionPathSteps = updatedException.path_and_schedule.path.map((_, i) =>
+        computeBasePathStep(updatedException.path_and_schedule!, i)
+      );
+      if (
+        originalPacedTrainPathSteps.length === exceptionPathSteps.length &&
+        originalPacedTrainPathSteps.every((pathStep, i) =>
+          isEqual(omit(pathStep, 'id'), omit(exceptionPathSteps[i], 'id'))
+        )
+      ) {
+        delete updatedException.path_and_schedule;
+      }
+    }
+
+    if (
+      exception.rolling_stock &&
+      isEqual(exception.rolling_stock.comfort, updatedPacedTrain.comfort) &&
+      isEqual(exception.rolling_stock.rolling_stock_name, updatedPacedTrain.rolling_stock_name)
+    ) {
+      delete exception.rolling_stock;
+    }
+
+    if (
+      exception.rolling_stock_category &&
+      isEqual(exception.rolling_stock_category.value, updatedPacedTrain.category)
+    ) {
+      delete updatedException.rolling_stock_category;
+    }
+
+    if (
+      exception.speed_limit_tag &&
+      isEqual(
+        exception.speed_limit_tag.value ?? null,
+        // speed limit tag is instanciated with null if not present when formating the item
+        updatedPacedTrain.speed_limit_tag ?? null
+      )
+    ) {
+      delete updatedException.speed_limit_tag;
+    }
+
+    // We do the check only for indexed occurrences because added exceptions should not have
+    // their start time reset
+    if (exception.start_time && exception.occurrence_index !== undefined) {
+      let originalStartTimeToTest = new Date(updatedPacedTrain.start_time);
+
+      const originalPacedTrainInterval = Duration.parse(updatedPacedTrain.paced.interval);
+      originalStartTimeToTest = dayjs(originalStartTimeToTest)
+        .add(exception.occurrence_index * originalPacedTrainInterval.ms, 'ms')
+        .toDate();
```

Comment:
You can simplify a bit like this:
```suggestion
      const originalStartTimeToTest = dayjs(new Date(updatedPacedTrain.start_time))
        .add(exception.occurrence_index * Duration.parse(updatedPacedTrain.paced.interval).ms, 'ms')
        .toDate();
```

Reply:
Don't you think it is a bit painful to read ? :)

Reply:
Done